source/amazon/troubleshooting.rst

```diff
@@ -0,0 +1,113 @@
+
+1. Module starting:
+
+.. code-block:: console
+
+    2018/01/12 18:47:09 wazuh-modulesd:aws-cloudtrail: INFO: Module AWS-CloudTrail started
+
+
+2. Scheduled scan:
+
+.. code-block:: console
+
+    2018/01/12 18:49:10 wazuh-modulesd:aws-cloudtrail: INFO: Fetching logs started
+    2018/01/12 18:49:11 wazuh-modulesd:aws-cloudtrail: INFO: Fetching logs finished.
+
+
+Troubleshooting
+---------------
+
+1. Errors in ossec.log
+
+When an error occurs when trying to collect and parse logs for a CloudTrail, the ossec.log will output an error such as below:
+
+The number is the AWS Account ID provided for the CloudTrail, and the name in the parenthesis is the AWS Account Alias (if provided).
+
+.. code-block:: console
+
+    2018/06/28 14:31:09 wazuh-modulesd:aws-cloudtrail: WARNING: CloudTrail: 012345678901(Prod)  -  Returned exit code 3.
+    2018/06/28 14:31:09 wazuh-modulesd:aws-cloudtrail: WARNING: CloudTrail: 012345678901(Prod)  -  Invalid credentials to access S3 Bucket
+
+The exit codes are as follows:
+
+
++-------+------------------------------------------------------+
+| Code  | Description                                          |
++-------+------------------------------------------------------+
+| 1     | Unknown error                                        |
++-------+------------------------------------------------------+
+| 2     | SIGINT                                               |
++-------+------------------------------------------------------+
+| 3     | Invalid credentials to access S3 bucket              |
++-------+------------------------------------------------------+
+| 4     | boto3 module missing                                 |
++-------+------------------------------------------------------+
+| 5     | Unexpected error accessing SQLite DB                 |
++-------+------------------------------------------------------+
+| 6     | Unable to create SQLite DB                           |
++-------+------------------------------------------------------+
+| 7     | Unexpected error querying/working with objects in S3 |
++-------+------------------------------------------------------+
+| 8     | Failed to decompress file                            |
++-------+------------------------------------------------------+
+| 9     | Failed to parse file                                 |
++-------+------------------------------------------------------+
+| 10    | Failed to execute DB cleanup                         |
++-------+------------------------------------------------------+
+| 11    | Unable to connect to Wazuh                           |
++-------+------------------------------------------------------+
+
+
+
+2. Debugging configuration:
+
+If you are unable to determine the issues from the ossec.log, you can run the modules in debug mode.  With Wazuh running, stop the moduled
+
+.. code-block:: console
+
+    # pkill wazuh-modulesd
```

Comment:
This is a valid option, but not the only one. The debug level can be set in the [`/etc/internal_options.conf` file](https://documentation.wazuh.com/current/user-manual/reference/internal-options.html?highlight=internal_options#wazuh-modules) using the `wazuh_modules.debug` option. I don't know if it's worth it to explain or it will make this more difficult to understand.

Reply:
IMO, I think a general troubleshooting section would be helpful that is more abstracted out from this individual wodle.

Reply:
You're right :).